### ISSUE TITLE
yaml display using IPython.display.Markdown works JupyterLab but not Voila

### ISSUE THREAD

**Issue by jgunstone** [voila-dashboards/voila#1159]
likely related too:
https://github.com/voila-dashboards/voila/issues/873
https://github.com/voila-dashboards/voila/issues/166

## Description

display markdown yaml works in JupyterLab but not in Voila

## Reproduce

```py
from IPython.display import Markdown, clear_output
from ipywidgets import ToggleButton, Output
import json
import yaml

STR_JSON="""{"menu": {
  "id": "file",
  "value": "File",
  "popup": {
    "menuitem": [
      {"value": "New", "onclick": "CreateNewDoc()"},
      {"value": "Open", "onclick": "OpenDoc()"},
      {"value": "Close", "onclick": "CloseDoc()"}
    ]
  }
}}"""


def _on_change(change):
    with out:
        clear_output()
        if as_yaml.value:
            s = yaml.dump(parsed, indent=2)
            display(Markdown("YAML displays in JupyterLab but not in Voila")) 
            display(Markdown("\n```yaml\n" + s + "\n```")) 
        else:
            s = json.dumps(parsed, indent=2)
            display(Markdown("\n```Python\n" + s + "\n```"))
            
parsed = json.loads(STR_JSON)            
out = Output()
as_yaml = ToggleButton(description='as_yaml?')
as_yaml.observe(_on_change, names='value')
as_yaml.value=True

display(as_yaml)
display(out)
```

![image](https://user-images.githubusercontent.com/21370980/173842029-3d2be270-d814-4026-a605-a5a6f916e645.png)

## Expected behavior

Expect what happens in JuptyerLab

## Context

- voila version  = 0.3.5
- Operating System and version: Ubuntu2004
- Browser and version: Chrome

### If using JupyterLab

<details><summary>Installed Labextensions</summary>
<pre>
(ipyautoui) jovyan@PORT4028:/mnt/c/engDev/git_mf/ipyautoui$ jupyter labextension list
JupyterLab v3.2.1
/home/jovyan/miniconda3/envs/ipyautoui/share/jupyter/labextensions
        bqplot v0.5.33 enabled OK (python, bqplot)
        jupyter-vuetify v1.8.2 enabled OK
        ipydatagrid v1.1.8 enabled OK
        jupyterlab-plotly v5.1.0 enabled OK
        nbdime-jupyterlab v2.1.1 enabled OK
        ipysheet v0.5.0 enabled OK (python, ipysheet)
        jupyter-vue v1.7.0 enabled OK
        jupyterlab-jupytext v1.3.4 enabled OK (python, jupytext)
        jupyterlab-unfold v0.2.2 enabled OK (python, jupyterlab-unfold)
        @jupyter-widgets/jupyterlab-manager v3.0.0 enabled OK (python, jupyterlab_widgets)
        @voila-dashboards/jupyterlab-preview v2.1.0 enabled OK (python, voila)
        @jupyterlab/git v0.37.1 enabled OK (python, jupyterlab-git)

Other labextensions (built into JupyterLab)
   app dir: /home/jovyan/miniconda3/envs/ipyautoui/share/jupyter/lab
</pre>
</details>

**Comment by martinRenou** [comment]
You cannot display new outputs in Voila, you'll need to embed your Markdown in an output widget for this to work, see similar issue https://github.com/voila-dashboards/voila/issues/1145

**Comment by jgunstone** [comment]
i'm not sure that was the issue... 

i tried the below, i.e. loading all the markdown into Output widgets initially, which still didn't work:  

(i may have misunderstood your point here?) 

```py

from IPython.display import Markdown, clear_output
from ipywidgets import ToggleButton, Output
import json
import yaml

STR_JSON="""{"menu": {
  "id": "file",
  "value": "File",
  "popup": {
    "menuitem": [
      {"value": "New", "onclick": "CreateNewDoc()"},
      {"value": "Open", "onclick": "OpenDoc()"},
      {"value": "Close", "onclick": "CloseDoc()"}
    ]
  }
}}"""

PARSED = json.loads(STR_JSON)  
OUT1 = Output()

with OUT1:
    s = yaml.dump(PARSED, indent=2)
    display(Markdown("YAML displays in JupyterLab but not in Voila")) 
    display(Markdown("\n```yaml\n" + s + "\n```")) 

    
OUT2 = Output()
with OUT2:
    s = json.dumps(PARSED, indent=2)
    display(Markdown("\n```Python\n" + s + "\n```"))

def _on_change(change):
    with out:
        clear_output()
        if as_yaml.value:
            display(OUT1)
        else:
            display(OUT2)
            
            
out = Output()
as_yaml = ToggleButton(description='as_yaml?')
as_yaml.observe(_on_change, names='value')
as_yaml.value=True

display(as_yaml)
display(out)
```


in the end I just used the rich console rather the IPython display system, which worked in both JupyterLab and Voila: 

```py
from IPython.display import  clear_output #Markdown,
from ipywidgets import ToggleButton, Output
import json
import yaml

from rich.console import Console, COLOR_SYSTEMS
from rich.markdown import Markdown


STR_JSON="""{"menu": {
  "id": "file",
  "value": "File",
  "popup": {
    "menuitem": [
      {"value": "New", "onclick": "CreateNewDoc()"},
      {"value": "Open", "onclick": "OpenDoc()"},
      {"value": "Close", "onclick": "CloseDoc()"}
    ]
  }
}}"""

PARSED = json.loads(STR_JSON)  
OUT1 = Output()

with OUT1:
    s = yaml.dump(PARSED, indent=2)
    display(Markdown("YAML displays in JupyterLab and Voila using the rich console")) 
    console = Console(style="white")
    console.print(Markdown("\n```yaml\n" + s + "\n```"))
    

    
OUT2 = Output()
with OUT2:
    s = json.dumps(PARSED, indent=2)
    display(Markdown("\n```Python\n" + s + "\n```"))

def _on_change(change):
    with out:
        clear_output()
        if as_yaml.value:
            display(OUT1)
        else:
            display(OUT2)
            
            
out = Output()
as_yaml = ToggleButton(description='as_yaml?')
as_yaml.observe(_on_change, names='value')
as_yaml.value=True

display(as_yaml)
display(out)
```

**Comment by martinRenou** [comment]
Sorry I read a bit fast your initial issue. There might be something else here that I missed. I'll have a deeper look.

**Comment by jgunstone** [comment]
this is the javascript console

![image](https://github.com/voila-dashboards/voila/assets/21370980/dd8c6241-bbad-4e8f-a5e8-79937a78e80b)

**Comment by martinRenou** [comment]
Would you be able to try with Voila v0.5.0a4? That may just work 🤞🏽

**Comment by martinRenou** [comment]
This should be fixed by #1353
